cppcheck reports some issues

**Description of the problem**

```
src/gmt_api.c:11722:4: error: Common realloc mistake: 'info' nulled but not freed upon failure [memleakOnRealloc]
   info = realloc (info, n_alloc * sizeof (struct GMT_RESOURCE));
   ^
src/gmt_api.c:11761:6: error: Common realloc mistake: 'info' nulled but not freed upon failure [memleakOnRealloc]
     info = realloc (info, n_alloc * sizeof (struct GMT_RESOURCE));
     ^
src/gmt_api.c:11771:36: error: Common realloc mistake: 'info' nulled but not freed upon failure [memleakOnRealloc]
 if (n_items && n_items < n_alloc) info = realloc (info, n_items * sizeof (struct GMT_RESOURCE));



src/gmt_shore.c:1567:64: error: Uninitialized variable: start [uninitvar]
   GMT->current.plot.n = gmt_map_truncate (GMT, xtmp, ytmp, n, start, -1);
                                                               ^
src/gmt_shore.c:1556:91: note: Assuming condition is false
  if ((n = (unsigned int)gmt_clip_to_map (GMT, p[k].lon, p[k].lat, p[k].n, &xtmp, &ytmp)) == 0) { /* Completely outside */
                                                                                          ^
src/gmt_shore.c:1567:64: note: Uninitialized variable: start
   GMT->current.plot.n = gmt_map_truncate (GMT, xtmp, ytmp, n, start, -1);



src/mgd77/cm4_functions.c:352:6: error: Common realloc mistake: 'dstx' nulled but not freed upon failure [memleakOnRealloc]
     dstx = realloc(dstx, (size_t)(n_Dst_rows * 24) * sizeof(double));


```

--- a/src/gmt_api.c
+++ b/src/gmt_api.c
@@ -11591,7 +11591,10 @@ struct GMT_RESOURCE * GMT_Encode_Options (void *V_API, const char *module_name, 
 		return_null (NULL, GMT_ONLY_ONE_ALLOWED);	/* Too many output objects */
 	}
 	n_alloc = n_keys;	/* Initial number of allocations */
-	info = calloc (n_alloc, sizeof (struct GMT_RESOURCE));
+	if ((info = gmt_M_memory (API->GMT, NULL, n_alloc, struct GMT_RESOURCE)) == NULL) {
+		GMT_Report (API, GMT_MSG_ERROR, "GMT_Encode_Options: Unable to allocate GMT_RESOURCE array\n");
+		return_null (NULL, GMT_MEMORY_ERROR);
+	}
 
 	if (!strncmp (module, "psrose", 6U) && (opt = GMT_Find_Option (API, 'E', *head)) && strcmp (opt->arg, "m")) {
 		/* Giving any -E option but -Em means we have either input or output so must update the key accordingly */
@@ -11719,7 +11722,10 @@ struct GMT_RESOURCE * GMT_Encode_Options (void *V_API, const char *module_name, 
 			            opt->option, opt->arg, satisfy);
 		if (n_items == n_alloc) {
 			n_alloc <<= 1;
-			info = realloc (info, n_alloc * sizeof (struct GMT_RESOURCE));
+			if ((info = gmt_M_memory (API->GMT, info, n_alloc, struct GMT_RESOURCE)) == NULL) {
+				GMT_Report (API, GMT_MSG_ERROR, "GMT_Encode_Options: Unable to reallocate GMT_RESOURCE array\n");
+				return_null (NULL, GMT_MEMORY_ERROR);
+			}
 		}
 	}
 
@@ -11758,7 +11764,10 @@ struct GMT_RESOURCE * GMT_Encode_Options (void *V_API, const char *module_name, 
 				if (direction == GMT_IN) n_in_added++;
 				if (n_items == n_alloc) {
 					n_alloc <<= 1;
-					info = realloc (info, n_alloc * sizeof (struct GMT_RESOURCE));
+					if ((info = gmt_M_memory (API->GMT, info, n_alloc, struct GMT_RESOURCE)) == NULL) {
+						GMT_Report (API, GMT_MSG_ERROR, "GMT_Encode_Options: Unable to reallocate GMT_RESOURCE array\n");
+						return_null (NULL, GMT_MEMORY_ERROR);
+					}
 				}
 			}
 		}
@@ -11768,8 +11777,14 @@ struct GMT_RESOURCE * GMT_Encode_Options (void *V_API, const char *module_name, 
 	gmt_M_str_free (key);
 
 	/* Reallocate the information structure array or remove entirely if nothing given. */
-	if (n_items && n_items < n_alloc) info = realloc (info, n_items * sizeof (struct GMT_RESOURCE));
-	else if (n_items == 0) gmt_M_str_free (info);	/* No containers used */
+	if (n_items && n_items < n_alloc) {
+		if ((info = gmt_M_memory (API->GMT, info, n_items, struct GMT_RESOURCE)) == NULL) {
+			GMT_Report (API, GMT_MSG_ERROR, "GMT_Encode_Options: Unable to finalize size of GMT_RESOURCE array\n");
+			return_null (NULL, GMT_MEMORY_ERROR);
+		}
+	}
+	else if (n_items == 0)
+		gmt_M_free (API->GMT, info);	/* No containers used */
 
 	gmt_M_memset (nn, 4, unsigned int);
 	for (ku = 0; ku < n_items; ku++)	/* Count how many primary and secondary objects each for input and output */

--- a/src/gmt_shore.c
+++ b/src/gmt_shore.c
@@ -1537,7 +1537,7 @@ int gmt_prep_shore_polygons (struct GMT_CTRL *GMT, struct GMT_GSHHS_POL **p_old,
 	 */
 
 	unsigned int k, np_new, n, n_use;
-	uint64_t start;
+	uint64_t start = 0;
 	bool close;
 	size_t n_alloc;
 	double *xtmp = NULL, *ytmp = NULL;

--- a/src/mgd77/cm4_functions.c
+++ b/src/mgd77/cm4_functions.c
@@ -158,6 +158,7 @@ int MGD77_cm4field (struct GMT_CTRL *GMT, struct MGD77_CM4 *Ctrl, double *p_lon,
 	double *ecto = NULL;		/* was [16416] */
 	double *hyto = NULL;		/* was [49248] */
 	char line[GMT_BUFSIZ] = {""}, *c_unused = NULL;
+	void *ptr = NULL;
 
 	FILE *fp;
 	gmt_M_unused(GMT);
@@ -349,7 +350,11 @@ int MGD77_cm4field (struct GMT_CTRL *GMT, struct MGD77_CM4 *Ctrl, double *p_lon,
 				k = (jmjd - mjdl) * 24;
 				if (k >= n_Dst_rows) {
 					n_Dst_rows += (100 * 24);
-					dstx = realloc(dstx, (size_t)(n_Dst_rows * 24) * sizeof(double));
+					ptr = (void *)dstx;
+					if ((dstx = realloc(dstx, (size_t)(n_Dst_rows * 24) * sizeof(double))) == NULL) {
+						gmt_M_str_free (ptr);
+						return 1;
+					}
 				}
 				for (j = 0; j < 24; ++j)
 					dstx[k + j] = (double)jdst[j];
@@ -358,8 +363,13 @@ int MGD77_cm4field (struct GMT_CTRL *GMT, struct MGD77_CM4 *Ctrl, double *p_lon,
 			fclose(fp);
 			mjdh = jmjd;
 		}
-		if (Ctrl->CM4_DATA.n_times > 1)	/* Need to re-allocate memory for all n_times in dst array */
-			Ctrl->CM4_D.dst = realloc(Ctrl->CM4_D.dst, (size_t)(Ctrl->CM4_DATA.n_times) * sizeof(double));
+		if (Ctrl->CM4_DATA.n_times > 1)	{	/* Need to re-allocate memory for all n_times in dst array */
+			ptr = (void *)Ctrl->CM4_D.dst;
+			if ((Ctrl->CM4_D.dst = realloc(Ctrl->CM4_D.dst, (size_t)(Ctrl->CM4_DATA.n_times) * sizeof(double))) == NULL) {
+				gmt_M_str_free (ptr);
+				return 1;
+			}
+		}
 
 		/* Get only one dst first so that we can test (and abort if needed) if date is out of bounds */
 		if (dstx) Ctrl->CM4_D.dst[0] = intdst(mjdl, mjdh, mjdy[0], msec[0], dstx, &cerr);

